File: 2_day/线程状态的膜/线程状态的膜/ViewController.m
//
//  ViewController.m
//  线程状态的膜
//
//  Created by 张正雨 on 2017/5/21.
//  Copyright © 2017年 张正雨. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    [NSThread detachNewThreadSelector:@selector(run:) toTarget:self withObject:@"abc"];
}
- (void)run:(id)obj{
    NSLog(@"---run---\n%@\n%@",[NSThread currentThread],obj);
    
    [NSThread sleepForTimeInterval:2];
    
    NSLog(@"再次执行");
    
    NSLog(@"%@",[NSDate distantFuture]);
    
    //强制退出线程
    for (int i = 0 ; i < 100000; i++) {
        if (i == 199) {
            NSLog(@"线程退出");
            [NSThread exit];
        }
    }
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 1_day/swift可选项/swift可选项/ViewController.swift
//
//  ViewController.swift
//  swift可选项
//
//  Created by 张正雨 on 2017/5/16.
//  Copyright © 2017年 张正雨. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        demo()
        demo1()
    }

    func demo() {
        let x:Optional = 10
        let y: Int? = nil
        
        print(x)
        print(y)
        //对nil进行解包会崩溃
//        print(x!+y!)
    }
    
    func demo1() {
        let y:Int?
        
        y = 20
//        y = 30 //let定义的常量，不能改变
        
        print(y)
        
        //var的可选项值默认为nil
        var x:Int?
        x = 10
        x = 100
        print(x!)
        
    }
}

